### DISCUSSION TITLE
Run the cli for a react app behinds authentication

### DISCUSSION THREAD

**Discussion by yixuanhou** [GoogleChrome/lighthouse-ci#946] [Q&A]
Hey, I'm setting up the lighthouse-ci for a react app that behinds authentication ( which uses @azure/msal-react). The app redirects users to the MS login page and redirect users back to the app once authenticated. 

According to the documentation, I've set up a puppeteer-script.js. However, when the lighthouse scans the app, it still gets stuck in the microsoft login page. And the lighthouse report says `The page may not be loading as expected because your test URL (http://localhost:61337/index.html) was redirected to https://login.microsoftonline.com/cde6*********. Try testing the second URL directly.` 

I passed the headless:false option into the puppeteer, but there's no chrome launched so hard to debug to verify if the puppeteer script works or not. Has anyone had experience with setting this up?
```
module.exports = async (browser, context) => {
  const page = await browser.newPage();

  await page.goto(context.url);

  await page.waitForSelector('input[type="email"]');

  await page.type('body input[type="email"]', email);
  console.log("email typed");
  await page.click('body input[type="submit"]');
  await page.type('body input[type="password"]', pwd);
  console.log("password typed");
  await page.click('body input[type="submit"]');

  await page.click('body input[type="submit"]');
};
```

The below is my lighthouserc.js

```
module.exports = {
  ci: {
    collect: {
      staticDistDir: "./build",
      puppeteerScript: "./puppeteer-script.js",
      puppeteerLaunchOptions: {
        headless: false,
        slowMo: 100,
      },
      settings: {
        disableStorageReset: true,
      },
      numberOfRuns: 1,
    },
    assert: {
      preset: "lighthouse:recommended",
    },
  },
};
```

![Screenshot 2023-10-08 at 8 30 28 PM](https://github.com/GoogleChrome/lighthouse-ci/assets/60414595/1563da9f-4dfa-4c20-88b7-c10962c143c4)

**Comment by knownasilya** [comment]
Seeing a similar issue, we are using cognito and our login is on a different subdomain from our app. After the puppeteer script runs the lighthouse scan happens on the `auth.<...>` instead of the original `app.<....>` domain. Is there a way to rerun the original url after the auth happens? Seems like changing urls in the puppeteer script doesn't change anything on this front.
